2/contrib/s2i/run

```diff
@@ -225,7 +223,7 @@ fi
 
 # if `docker run` first argument start with `--` the user is passing jenkins launcher arguments
 if [[ $# -lt 1 ]] || [[ "$1" == "--"* ]]; then
-   exec java $JAVA_GC_OPTS $JAVA_INITIAL_HEAP_PARAM $JAVA_MAX_HEAP_PARAM -Duser.home=${HOME} $JAVA_CORE_LIMIT $JAVA_DIAGNOSTICS $JAVA_OPTS -Dfile.encoding=UTF8 -jar /usr/lib/jenkins/jenkins.war $JENKINS_OPTS $JENKINS_ACCESSLOG "$@"
+   exec java -XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap $JAVA_GC_OPTS $JAVA_INITIAL_HEAP_PARAM $JAVA_MAX_HEAP_PARAM -Duser.home=${HOME} $JAVA_CORE_LIMIT $JAVA_DIAGNOSTICS $JAVA_OPTS -Dsun.zip.disableMemoryMapping=true -Dfile.encoding=UTF8 -jar /usr/lib/jenkins/jenkins.war $JENKINS_OPTS $JENKINS_ACCESSLOG "$@"
```

Comment:
Yeah we had heard that the experimental options re: cgroups were coming, but they were still not available at that time

Cool deal if it is now - thanks.

Reply:
while we're messing with this i think we should ensure that all the JVM args being passed in can be overwritten by the user in case this causes problems.  e.g.:

```
if [ -z JVM_ARGS ]; then
  JVM_ARGS =  -XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap $JAVA_GC_OPTS $JAVA_INITIAL_HEAP_PARAM $JAVA_MAX_HEAP_PARAM -Duser.home=${HOME} $JAVA_CORE_LIMIT $JAVA_DIAGNOSTICS $JAVA_OPTS -Dsun.zip.disableMemoryMapping=true -Dfile.encoding=UTF8
fi

exec java $JVM_ARGS -jar /usr/lib/jenkins/jenkins.war $JENKINS_OPTS $JENKINS_ACCESSLOG "$@"
```

also not sure what UseCGroupMemoryLimitForHeap accomplishes here since we're explicitly setting the heap size anyway.

Reply:
OK. I'll make the changes. @bparees There are some off-heap structures of the JVM which get sized correctly with this option. -Xmx is heap only. The JVM is not just heap. I've tried to explain this in the initial PR comment.

Reply:
Sorry, did not read the PR comment closely :(